lib/pages/goal_page.dart

```diff
@@ -3,102 +3,529 @@ import 'package:linear/model/goal.dart';
 import 'package:linear/pages/common_widgets/navbar.dart';
 import 'package:linear/pages/common_widgets/goal_widget.dart';
 import 'package:linear/util/apis.dart' as api;
-
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+import 'package:linear/model/user.dart';
+import 'package:linear/pages/profile_page/profile_page.dart';
+import 'package:linear/model/goal.dart';
+ 
+// ignore: must_be_immutable
 class GoalPage extends StatefulWidget {
-  const GoalPage({super.key});
-
-  @override
-  State<GoalPage> createState() => GoalPageState();
+ const GoalPage({super.key});
+ 
+ 
+ @override
+ State<GoalPage> createState() => GoalPageState();
 }
-
+ 
 class GoalPageState extends State<GoalPage> {
-  List<dynamic> _goals = [];
+ 
+    List<dynamic> _allGoals= [];
+    List<dynamic> _goals = [];
+    String sortingBy = "Newest";
+    String filteringBy= "Open";
+    bool isSorting = true;
 
-  bool _isLoading = true;
-  bool _isErrorFetchingGoals = false;
+    TextEditingController searchInput = TextEditingController();
 
-  @override
-  void initState() {
-    super.initState();
-    fetchGoals();
-  }
+  List<dynamic> _searchResults = [];
+  bool _hasPerformedSearch = false;
+   bool _isSearching = false;
+ 
+ bool _isLoading = true;
```

Comment:
`isLoading` should be initialized to `false` and set to `true` in the actual function that does the loading.